playbooks/files/promtail/promtail.yml.j2

```
@@ -0,0 +1,21 @@
+# vim:ts=2:sw=2:et:ai:sts=2
+---
+server:
+  http_listen_port: 9080
+  grpc_listen_port: 0
+
+positions:
+  filename: /var/lib/promtail/positions.yaml
+
+clients:
+  - url: {{ promtail__push_url }}
+
+scrape_configs:
+  - job_name: system
+    static_configs:
+      - targets:
+          - localhost
+        labels:
+          job: varlogs
+          __path__: /var/log/*log
+          stream: stdout
```

The `targets: [localhost]` block is not necessary, AFAIK. And I'd use the same name for `job_name` and the `job` label.

More imporantly, this section will always need customisation, to be able to adapt to each host needs. But it is also incredibly complicated, so I think it is better if you just take an arbitrary  list of dicts, and with var aggregation later you can add configurations modularly


```suggestion
scrape_configs:
  {{ promtail__scrape_configs | to_nice_yaml(indent=2) | indent(2) }}
```

And variables like:
```yaml
promtail__scrape_configs: |-
  {{ query('vars', *(query('varnames', '^promtail__scrape_config__.*') | select | sort)) }}

promtail__scrape_config__varlogs:
  job_name: varlogs
  static_configs:
    - abels:
        job: varlogs
        __path__: /var/log/*log
        stream: stdout

```

---

Applied the suggestion. But I have added `promtail__scrape_configs` herein this PR (to generic playbook) and promtail__scrape_config__varlogs to ihive PR (https://github.com/HiveNetCode/hive-compute-sysadmin/pull/107) since it is particular to the client. Is that correct? PTAL.

---

In this case, I think I would leave both variables here, since that is a reasonable default for any linux host, and it can be overridden by the user.

---

Ok, I have taken back `promtail__scrape_config__varlogs` to the generic playbook.